Mathlib/Data/Setoid/Basic.lean

```diff
@@ -442,16 +442,16 @@ def sigmaQuotientEquivOfLe {r s : Setoid α} (hle : r ≤ s) :
 end Setoid
 
 @[simp]
-theorem Quotient.subsingleton_iff {s : Setoid α} : Subsingleton (Quotient s) ↔ s = ⊤ := by
-  simp only [_root_.subsingleton_iff, eq_top_iff, Setoid.le_def, Setoid.top_def, Pi.top_apply]
-  refine Quotient.mk'_surjective.forall.trans (forall_congr' fun a => ?_)
-  refine Quotient.mk'_surjective.forall.trans (forall_congr' fun b => ?_)
-  simp_rw [Prop.top_eq_true, true_implies, Quotient.eq']
+nonrec theorem Quotient.subsingleton_iff {s : Setoid α} : Subsingleton (Quotient s) ↔ s = ⊤ := by
```

Comment:
I know it's easy to prove, but for simp we should add
```lean4
@[simp]
nonrec theorem Quotient.nontrivial_iff {s : Setoid α} : Subsingleton (Quotient s) ↔ s ≠ ⊤ := by
```
especially as this also golfs your proof below